### DISCUSSION TITLE
Any way to mimic `BuildInfo`?

### DISCUSSION THREAD

**Discussion by ckipp01** [VirtusLab/scala-cli#2051] [General]
I have a use case where I'm building a CLI app. A lot of times with apps like this you want to make sure you can display the version of the app via a command, and many times the value of that version is persisted utilizing a form of `BuildInfo` where based on git, or something else a file is generated and used for compilation. Is there any way to mimic something like this with `scala-cli`? All the solutions I'm coming up with request running `scala-cli` to generate a file first, and then running it again on your main project to utilize that created file. However, it'd be great if this could just be done in a single step somehow. Any ideas?

**Comment by ckipp01** [comment]
As a temporary work around what I have now is a Makefile task that just calls

```
generate-build-info:
	scala-cli run scripts/generate-build-info.scala

compile:
	make generate-build-info
	scala-cli compile skan
```

That script grabs the current version and sticks it in the `.scala-build/BuildInfo.scala` and then inside of my `project.scala` I have:

```
//> using file "../.scala-build/BuildInfo.scala"
```

Does it look pretty? Nope, but it works. So I'll use this until I figured out a better way to do this 😆

**Reply by bishabosha** [reply]
there's `scala --power export --json` which is pretty useful for grabbing any build info needed, and in the newest release it will print to std-out. Do you use that?

also there is this issue for source generators (unresolved): https://github.com/VirtusLab/scala-cli/issues/610

**Reply by ckipp01** [reply]
I should have mentioned that. I did look at using:

```//> using publish.computeVersion "git:tag"```

But this doesn't actually export anything related to the version it computes when you do the `export --json`.

> also there is this issue for source generators (unresolved): https://github.com/VirtusLab/scala-cli/issues/610

Oooo, this would be great.
